occupancy_sensing.cpp

```diff
@@ -0,0 +1,267 @@
+#include "de_web_plugin.h"
+#include "de_web_plugin_private.h"
+
+#define OCCUPIED_STATE                  0x0000
+#define OCCUPIED_TO_UNOCCUPIED_DELAY    0x0010
+#define HUE_SENSITIVITY                 0x0030
+#define HUE_SENSITIVITY_MAX             0x0031
+
+/*! Handle packets related to the ZCL occupancy sensing cluster.
+    \param ind the APS level data indication containing the ZCL packet
+    \param zclFrame the actual ZCL frame which holds the occupancy sensing cluster command or attribute
+ */
+void DeRestPluginPrivate::handleOccupancySensingClusterIndication(const deCONZ::ApsDataIndication &ind, const deCONZ::ZclFrame &zclFrame)
+{
+    if (zclFrame.isDefaultResponse())
+    {
+        return;
+    }
+
+    Sensor *sensor = getSensorNodeForAddressEndpointAndCluster(ind.srcAddress(), ind.srcEndpoint(), OCCUPANCY_SENSING_CLUSTER_ID );
+
+    if (!sensor)
+    {
+        DBG_Printf(DBG_INFO, "No presence sensor found for 0x%016llX, endpoint: 0x%02X\n", ind.srcAddress().ext(), ind.srcEndpoint());
+        return;
+    }
+
+    QDataStream stream(zclFrame.payload());
+    stream.setByteOrder(QDataStream::LittleEndian);
+
+    bool isReadAttr = false;
+    bool isReporting = false;
+    if (zclFrame.isProfileWideCommand() && zclFrame.commandId() == deCONZ::ZclReadAttributesResponseId)
+    {
+        isReadAttr = true;
+    }
+    if (zclFrame.isProfileWideCommand() && zclFrame.commandId() == deCONZ::ZclReportAttributesId)
+    {
+        isReporting = true;
+    }
+
+    // Read ZCL reporting and ZCL Read Attributes Response
+    if (isReadAttr || isReporting)
```

Comment:
```cpp
    if (zclFrame.isProfileWideCommand() && zclFrame.commandId() == deCONZ::ZclReadAttributesResponseId)
    {
        isReadAttr = true;
    }
    else if (zclFrame.isProfileWideCommand() && zclFrame.commandId() == deCONZ::ZclReportAttributesId)
    {
        isReporting = true;
    }
    else
    {
        return; // neigher ZCL Report nor ZCL Read Attributes Response
    }
```

With that we can get rid of the huge *if* scope.

```diff
-    // Read ZCL reporting and ZCL Read Attributes Response
-    if (isReadAttr || isReporting)
-    {
-    }
```

Reply:
Sure